### DISCUSSION TITLE
Hex unicode

### DISCUSSION THREAD

**Discussion by andrewbaxter** [elkowar/eww#1095] [Q&A]
I couldn't find any mentions in the documentation, and looking at various example repos everyone pastes strange unicode codepoints into their configs directly, so I assume that the string literals in yuck simply do not support unicode escapes.

There was a PR that introduced `hex_decode` for this reason but it looked like that particular function got excised before merging.

In my browser and VS Code the pasted unicode characters look like gibberish arabic, Chinese, etc. I'd very much prefer using unicode directly since I don't want to install and configure all the icon fonts in my editor (not to mention browser), and dealing with a simple ascii sequence like `\u1234` is easy to deal with.

Either having unicode escapes supported in yuck or a helper method like `hex_decode` would be awesome.

**Comment by w-lfchen** [comment]
i might be able to look into this if you point me to the specific pr.

the unicode symbols are most likely icons from [nerdfonts](https://www.nerdfonts.com/), a project that extends existing fonts with their set of icons. since they're monospaced, you can use them in vscode as well, and you can look up symbols and copy them [here](https://www.nerdfonts.com/cheat-sheet)

**Reply by andrewbaxter** [reply]
I think this was the MR: https://github.com/elkowar/eww/pull/407

ATM I've moved away from this project so this isn't something I actively need.

**Comment by MartyMcFlyInTheSky** [comment]
I'm still interested in this for the same exact reasons originally stated.

**Comment by w-lfchen** [comment]
looked into this, this feature exists (though we use an [outdated crate](https://github.com/saghm/unescape-rs), i'll into improving things here if possible)
however, due to the way escape sequences are treated, you'll need two backslashes:
`"\\u0038"` will yield `8`

see #1117 as well
